src/main/java/wtf/meier/tariff/interpreter/model/RentalPeriod.kt

```diff
@@ -0,0 +1,17 @@
+package wtf.meier.tariff.interpreter.model
+
+import wtf.meier.tariff.interpreter.model.rate.RateCalculator
+import wtf.meier.tariff.interpreter.extension.minus
+import java.time.Instant
+import java.util.concurrent.TimeUnit
+
+data class RentalPeriod(
+    val rentalStart: Instant = Instant.ofEpochMilli(0),
+    val rentalEnd: Instant,
+    val calculatedStart: Instant = rentalStart,
+    val calculatedEnd: Instant = rentalEnd,
+    val positions: MutableList<RateCalculator.CalculatedPrice> = mutableListOf()
```

Comment:
We shouldn't make this a mutable list. All  data classes should be immutable

Reply:
Those will always just be the positions for the goodwill offsetting, right? If so, let us make this clear with naming

Reply:
Right, and if we have only one goodwill and no list, we can change the value to chargedGoodwill